### PR TITLE
Configurable timeout for worker ASG creation

### PR DESCRIPTION
**Pull request by subos2008**
# PR o'clock

## Description

Please explain the changes you made here and link to any relevant issues.

### Checklist

- [x] `terraform fmt` and `terraform validate` both work from the root and `examples/*` directories
- [x] CI tests are passing
- [x] I've added my change to CHANGELOG.md and highlighted any breaking changes
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

File: workers.tf [hunk 1]
```diff
@@ -12,6 +12,7 @@ resource "aws_autoscaling_group" "workers" {
       ]
     )
   )
+  wait_for_capacity_timeout = var.worker_asg_create_timeout
```
Comment (barryib) [inline workers.tf]: This should be defined per ASG. You can add `asg_wait_for_capacity_timeout` default in [`local.workers_group_defaults_defaults`](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/local.tf#L26-L31).

Comment (max-rocket-internet) [reply workers.tf]: Maybe having this setting per worker group is too much? Can you imagine wanting a timeout of say 10m for one worker group but 20m for another?

**Comment by max-rocket-internet** [comment]
Please add to `workers_launch_template.tf` also.

File: workers.tf (hunk 1 above)
Comment (barryib) [reply workers.tf]: Not really in my case. But it's more to be consistent with other ASG options.

And by putting into the workers_group_defaults_defaults, you can achieve both defaults for all ASG and custom timeout per ASG (for those people with weird use cases :)) for free.

**Comment by barryib** [comment]
> Please add to `workers_launch_template.tf` also.

nice catch.

File: workers.tf (hunk 1 above)
Comment (max-rocket-internet) [reply workers.tf]: OK cool, sounds good 🙂

**Comment by barryib** [comment]
@subos2008 are you still working on this ?

**Comment by sidorovdmitry** [comment]
Maybe it can be closed because of new EKS managed worker nodes.

**Comment by max-rocket-internet** [comment]
Will close due to inactivity. Feel free to reopen and resolve the conflicts and feedback.

**Pull request closed without merging**